test/stdlib/SmallString.swift

```diff
@@ -132,47 +54,6 @@ SmallStringTests.test("FitsInSmall") {
 
 }
 
-#if _runtime(_ObjC)
-SmallStringTests.test("Bridging") {
-  // Test bridging retains small string form
-  func bridge(_ small: _SmallString) -> String {
-    return String(_StringGuts(small))._bridgeToObjectiveCImpl() as! String
-  }
-  func runTestSmall(_ input: String) throws {
-    // Constructed through CF
-    guard let fromCocoaSmall = _SmallString(
-      _cocoaString: input as NSString
-    ) else {
-        throw "Didn't fit"
-    }
-    verifySmallString(fromCocoaSmall, input)
-    verifySmallString(fromCocoaSmall, bridge(fromCocoaSmall))
-  }
-
-  // Pass tests
-
-  if #available(macOS 10.10, iOS 9, *) {
-    expectDoesNotThrow({ try runTestSmall("abc") })
-    expectDoesNotThrow({ try runTestSmall("defghijk") })
-    expectDoesNotThrow({ try runTestSmall("aaaaaaaaaaa") })
-  } else {
-    // OS X 10.9, iOS 7/8 did not have tagged strings
-    expectThrows("Didn't fit", { try runTestSmall("abc") })
-    expectThrows("Didn't fit", { try runTestSmall("defghijk") })
-    expectThrows("Didn't fit", { try runTestSmall("aaaaaaaaaaa") })
-  }
-
-  // Fail tests
-  //
-  expectThrows("Didn't fit", { try runTestSmall("\u{0}") })
-  expectThrows("Didn't fit", { try runTestSmall("0123456789abcde") })
-  expectThrows("Didn't fit", { try runTestSmall("👨‍👦abcd") })
-  expectThrows("Didn't fit", { try runTestSmall("👨‍👦") })
-  expectThrows("Didn't fit", { try runTestSmall("👨‍👩‍👦") })
-  expectThrows("Didn't fit", { try runTestSmall("👨‍👦abcde") })
-}
-#endif
-
```

Comment:
Why are we splitting this up into multiple files and do a fake header-include mechanism? Would passing a `-D` help instead?

Reply:
Because SmallString.swift has an implicit Foundation dependency (if Foundation isn't loaded, tagged pointer strings are disabled), but also wants to run on Linux

Reply:
And why does that require us to split up the files? Many tests have Foundation dependencies, or at least portions of the test with that dependency, that we care about running on Linux.

Reply:
What's new is that we have to add -framework Foundation to the run line, and asking on Slack didn't turn up any good ways of conditionalizing that. If there are good ways of doing that I would love to not do all this hoop jumping.

Reply:
So, what about all our other tests that we want to run on Linux, such as https://github.com/apple/swift/blob/master/validation-test/stdlib/String.swift ?